src/target/linux/software_breakpoint.rs

```diff
@@ -0,0 +1,101 @@
+//! Bundles functionalities related to software breakpoints
+//! Software Breakpoints work by overwritting the target program's memory.
+//! replacing an instruction with one that causes a signal to be raised by the
+//! cpu.
+
+use nix::sys::ptrace;
+use nix::unistd::Pid;
+const INT3: libc::c_long = 0xcc;
+/*
+pub struct Breakpoint2 {
+    bp: Breakpoint,
+    callback: Box<dyn FnMut(i32, i32)>,
+}
+
+impl Breakpoint2 {
+    pub fn new(
+        addr: usize,
+        pid: Pid,
+        callback: Box<dyn FnMut(i32, i32)>,
+    ) -> Result<Self, BreakpointError> {
+        let bp = Breakpoint::new(addr, pid)?;
+        Ok(Breakpoint2 { bp, callback })
+    }
+
+    fn test() {
+        let cb = |i1, i2| {
+            i1 + i2;
+        };
+        Breakpoint2::new(123, nix::unistd::Pid::this(), Box::new(cb));
+    }
+}
+*/
+
+#[derive(Debug, Copy, Clone)]
+pub struct Breakpoint {
+    /// The address at which the debugger should insert this breakpoint
+    pub addr: usize,
+    /// The shadowed variable
+    pub(crate) shadow: i64,
```

Comment:
```suggestion
    pub(super) shadow: i64,
```

The doc comment is not clear to me.